pkg/vpcmodel/grouping.go

```diff
@@ -319,9 +320,15 @@ func (g *GroupConnLines) groupExternalAddressesForExplainability() error {
 	var res []*groupedConnLine
 	for _, rulesSrcDst := range *g.explain {
 		connStr := rulesSrcDst.conn.String() + semicolon
-		groupingStrKey := connStr + rulesSrcDst.rules.rulesEncode(g.config)
+		routingStr := ""
+		if rulesSrcDst.router != nil {
+			router := rulesSrcDst.router
+			routingStr = router.Name() + ";"
+		}
+		groupingStrKey := connStr + routingStr + rulesSrcDst.rules.rulesEncode(g.config)
```

Comment:
why not having `rulesEncode()` be called by `rulesSrcDst` and consider both the `router` and the `rules` from its own struct?

Reply:
done; I've also renamed the type `rulesSrcDst`  to `srcDstDetails`